src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView.java

```diff
@@ -606,11 +606,15 @@ private int triggerBuild(final AbstractProject<?, ?> triggerProject, final Abstr
         if (manualTrigger != null) {
             final Set<String> downstreamProjectsNames =
                     Sets.newHashSet(Splitter.on(",").trimResults().split(manualTrigger.getDownstreamProjectNames()));
-            if (downstreamProjectsNames.contains(project.getFullName())) {

```

Comment:
I cleaned the commit I did. Apologize about it.

Regarding your comments, I am not sure if I understand... Do you have any use case in which this change is not going to work or will break something? I would be happy to give a try.